File: code-coaches/hard/mathematics/mathematics.rb
def find_matching_expression(answer, expressions)
  expressions.each_with_index do |expression, i|
    result = eval(expression)
    return i if result == answer
  end
  return nil
end

def main
  answer = gets.chomp.to_i
  expressions = gets.chomp.split

  index = find_matching_expression(answer, expressions)
  if index
    puts "index #{index}"
  else
    puts "none"
  end
end

if __FILE__ == $0
  main
end

File: code-coaches/hard/mathematics/mathematics.py
def find_matching_expression(answer, expressions):
    for i, expression in enumerate(expressions):
        result = eval(expression)
        if result == answer:
            return i

    return None


def main():
    answer = int(input())
    expressions = input().split()

    index = find_matching_expression(answer, expressions)
    if index is not None:
        print(f"index {index}")
    else:
        print("none")


if __name__ == "__main__":
    main()
